plugins/by-name/langmapper/default.nix

```diff
@@ -0,0 +1,104 @@
+{
+  lib,
+  config,
+  ...
+}:
+let
+  inherit (lib)
+    mkEnableOption
+    mkOption
+    optional
+    types
+    mkIf
+    concatStringsSep
+    map
+    ;
+  inherit (lib.nixvim.lua) toLua;
+  inherit (lib.nixvim.defaultNullOpts) mkBool mkListOf;
+
+  mkKeyModes = mkListOf lib.nixvim.keymaps.modeType;
+  cfg = config.plugins.langmapper;
+in
+lib.nixvim.plugins.mkNeovimPlugin {
+  name = "langmapper";
+  package = "langmapper-nvim";
+  description = "A plugin that makes Neovim more friendly to non-English input methods";
+  maintainers = [ lib.maintainers.shved ];
+  settingsOptions = {
```

Comment:
Is it necessary to have these settings sub-options explicitly declared? And are you willing to keep them up-to-date if upstream changes?

Generally, in newer modules we tend to rely more on the freeform nature of `settings` and avoid explicitly declaring settings sub-options.